browsing-context.md

```diff
@@ -178,6 +178,8 @@ The current proposed API is a `document.loadingMode` object with:
 
 Envisioned usage is as follows:
 
+TODO: This example would likely want to use `document.prerendering` - do we have a example that demonstrates the need for additional granularity?
```

Comment:
I can work on this TODO after this is merged.

In particular, I'm now feeling that we should reframe this section to say something like:

- Use all these individual APIs!
- But, maybe it'd be a good idea to expose the underlying primitive that the user agent is using, i.e. the loading mode? If so, here's our proposal. We're not yet aware of any use cases though, so consider this on-hold.

Reply:
Totally open to hiding loading mode from the JS API if we think the use cases that matter are covered by narrower APIs, and adding it later on if we actually need it (and are better informed about what it should look like).

Reply:
I've been thinking of `document.prerendering` as being based on whether or not the document is being rendered in a prerendering browsing context. Assuming that's true, it's also an underlying primitive.

The loading-mode seems related but maybe orthogonal? Thought experiment: would it ever make sense to have an existing loading mode (other than `default`), or perhaps a new one in the future, that was used in a regular browsing context? Or using the 'default' loading mode in a prerendering browsing context (would we use that for same-origin prerender/portal)? This doesn't sound crazy to me so I think maybe `document.prerendering` isn't necessarily just sugar for `loadingMode != 'default'`